build.sh

```diff
@@ -53,7 +53,7 @@ RAN_CMAKE=0
 # If INSTALL_PREFIX is not set, check PREFIX, then check
 # CONDA_PREFIX, then fall back to install inside of $LIBRMM_BUILD_DIR
 INSTALL_PREFIX=${INSTALL_PREFIX:=${PREFIX:=${CONDA_PREFIX:=$LIBRMM_BUILD_DIR/install}}}
-export PARALLEL_LEVEL=${PARALLEL_LEVEL:-4}
+export PARALLEL_LEVEL=${PARALLEL_LEVEL:=$(nproc)}
```

Comment:
From the [Bash manual](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html):

> `${parameter:-word}`
>
> If *parameter* is unset or null, the expansion of *word* is substituted. Otherwise, the value of *parameter* is substituted.
>
> ```
> $ v=123
> $ echo ${v-unset}
> 123
> ```
>
> `${parameter:=word}`
>
> If *parameter* is unset or null, the expansion of *word* is assigned to *parameter*. The value of *parameter* is then substituted. Positional parameters and special parameters may not be assigned to in this way.
>
> ```
> $ var=
> $ : ${var:=DEFAULT}
> $ echo $var
> DEFAULT
> ```

Any particular reason why you've changed this from a substitution to a substitution-assignment?

Reply:
Note there's a footgun lurking here. If a runner sets `OMP_NUM_THREADS`, then `nproc` obeys that setting. You may wish to use `nproc --all` instead.

Reply:
PR updated with `nproc --all`, thank you